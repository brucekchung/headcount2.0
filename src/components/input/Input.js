import React, { Component } from 'react'
import PropTypes from 'prop-types'

class Input extends Component {
  constructor(props) {
    super(props)

    this.state = {
      value: ''
    }
  }

  handleChange = e => {
    const value = e.target.value

    this.setState({ value: value })
    this.props.handleSearch(value)
  }

  render() {
    return (
      <div className="controls">
        <input type="text" 
             value={ this.state.value } 
             placeholder="Search districts" 
             onChange={ this.handleChange } />
        <button onClick={ this.props.clearSelection }>Clear</button>
      </div>
    )
  }
}

Input.propTypes = {
  handleSearch: PropTypes.func.isRequired,
  state: PropTypes.shape({
    value: PropTypes.string.isRequired
  })
}

export default Input