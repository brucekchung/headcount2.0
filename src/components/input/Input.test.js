/* eslint-disable */
import React from 'react'
import Input from './Input'
import {shallow} from 'enzyme'

describe('Input', () => {
  let wrapper;
  const clearSelection = jest.fn()
  const handleSearch = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <Input handleSearch={handleSearch} clearSelection={clearSelection}/>
    )
  })

  it('should exist', () => {
    expect(wrapper).toBeDefined()
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have a default empty state.value', () => {
    const inst = wrapper.instance()

    expect(inst.state.value).toEqual('')
  })

  it('should update state on input field change', () => {
    const inst = wrapper.instance()
    expect(inst.state.value).toEqual('')
    const input = wrapper.find('input')
    input.simulate('change', {target: {value: 'abc'}})

    expect(inst.state.value).toEqual('abc')
  })
})


