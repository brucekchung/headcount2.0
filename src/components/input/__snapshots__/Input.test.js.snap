// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    clearSelection={[Function]}
    handleSearch={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          placeholder="Search districts"
          type="text"
          value=""
/>,
        <button
          onClick={[Function]}
>
          Clear
</button>,
      ],
      "className": "controls",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "Search districts",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Clear",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Clear",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            placeholder="Search districts"
            type="text"
            value=""
/>,
          <button
            onClick={[Function]}
>
            Clear
</button>,
        ],
        "className": "controls",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Search districts",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Clear",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Clear",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
